common/src/main/java/org/apache/celeborn/common/network/client/TransportClient.java

```diff
@@ -216,6 +216,23 @@ public ChannelFuture pushData(
       long pushDataTimeout,
       RpcResponseCallback callback,
       Runnable rpcSendoutCallback) {
+    Runnable rpcFailureCallback =
+        () -> {
+          try {
+            pushData.body().release();
+          } catch (Throwable e) {
+            logger.error("Error release buffer for PUSH_DATA request " + pushData.requestId, e);
```

Comment:
Here is a nit. I think you should use {} instead of + when logging.

Reply:
Done